scanner/matcher/updater/vuln/updater.go

```diff
@@ -185,6 +195,77 @@ func validate(opts Opts) error {
 	return nil
 }
 
+func (u *Updater) Import(ctx context.Context, in io.Reader) (err error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "matcher/updater/Updater.Import")
+	iter, iterErr := u.iterateFunc(in)
+	// For each update operation in the JSON blob file.
+	iter(func(op *driver.UpdateOperation, it jsonblob.RecordIter) bool {
+		var ops map[string][]driver.UpdateOperation
+		ops, err = u.store.GetUpdateOperations(ctx, op.Kind, op.Updater)
+		if err != nil {
+			return false
+		}
+		for _, o := range ops[op.Updater] {
+			// This only helps if updaters don't keep something that
+			// changes in the fingerprint.
+			if o.Fingerprint == op.Fingerprint {
+				zlog.Info(ctx).
+					Str("updater", op.Updater).
+					Msg("fingerprint match, skipping")
+				return true
+			}
+		}
+		var ref uuid.UUID
+		count := 0
+		switch op.Kind {
+		case driver.VulnerabilityKind:
+			ref, err = u.store.UpdateVulnerabilitiesIter(ctx, op.Updater, op.Fingerprint, func(yield func(*claircore.Vulnerability, error) bool) {
+				// For each vulnerability in the update operation.
+				it(func(v *claircore.Vulnerability, _ *driver.EnrichmentRecord) bool {
+					count++
+					// Offer one vulnerability to the datastore iterator.
+					return yield(v, nil)
```

Comment:
I don't think we `nil` check inside of `UpdaterVulnerabilitiesIter` (nor `UpdateEnrichmentsIter`). Should we add a check to be extra safe?

Reply:
The iterator contract is it should not return `nil` for vulnerability operation types, so `nil` here would mean a programmer's error, which is not recoverable; Ignoring this would create a bug where we would not insert some vulnerabilities, skip bundles, etc.